src/python/pants/backend/python/lint/pylint/rules.py

```diff
@@ -75,10 +77,14 @@ def generate_argv(source_files: SourceFiles, pylint: Pylint) -> Tuple[str, ...]:
 async def pylint_lint_partition(
     partition: PylintPartition, pylint: Pylint, first_party_plugins: PylintFirstPartyPlugins
 ) -> LintResult:
+    transitive_targets_get = Get(
+        TransitiveTargets, TransitiveTargetsRequest(fs.address for fs in partition.field_sets)
+    )
```

Comment:
AFAICT, this replaces one call to `Get(CoarsenedTargets, ..)` with a call to `Get(TransitiveTargets, ..)` per partition.

If the partitions overlap at all (which I think will happen when inner nodes in the graph have very compatible ICs, while the roots are more constrainted), I expect that that will be slower due to #11270, because each `TransitiveTargets` request will do redundant re-computation of the dependencies.

An alternative would be to replace `closure: FrozenOrderedSet[Target]` with `root_targets: CoarsenedTargets`, and then call `root_targets.closure()` (which is still a transitive walk, but is entirely on the stack as opposed to spread across many `@rule` calls).